format/midi/midi.go

```diff
@@ -33,17 +48,31 @@ func init() {
 	interp.RegisterFS(midiFS)
 }
 
+// decodeMIDI implements the MIDI file decoder.
+//
+// The decoder parses the file as a set of chunks, each comprising a 4 character tag
+// followed by a uint32 length field. The decoder parses the MTHd and MTrk MIDI chunks.
 func decodeMIDI(d *decode.D) any {
 	d.Endian = decode.BigEndian
 
 	// ... decode header
-	d.FieldStruct("header", decodeMThd)
+	format, _ := decodeMThd(d)
```

Comment:
Oh you also ran into the issue of wanting to use `d.Field...` but also return a value? i've thought about if generics could help but think i ran into some limitations

Reply:
&lt;grin&gt; I sat there for a whole minute wishing `d.FieldStruct` returned a value and then resigned myself to doing it the messy way.

Reply:
Yeah sadly a bit messy and methods are not allowed to have type params in go :(